From: SeongJae Park <sj@kernel.org>
Subject: mm/memory_hotplug: cleanup 'ret' variable usage in do_migrate_range()
Date: Wed, 15 Feb 2023 23:03:00 +0000

Because do_migrate_range() returns nothing, some 'ret' variable usages are
unnecessary.  Remove unnecessary usage and reduce its scope.

Link: https://lkml.kernel.org/r/20230215230300.61125-3-sj@kernel.org
Signed-off-by: SeongJae Park <sj@kernel.org>
Cc: David Hildenbrand <david@redhat.com>
Cc: Oscar Salvador <osalvador@suse.de>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/memory_hotplug.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

--- a/mm/memory_hotplug.c~mm-memory_hotplug-cleanup-ret-variable-usage-in-do_migrate_range
+++ a/mm/memory_hotplug.c
@@ -1624,7 +1624,6 @@ static void do_migrate_range(unsigned lo
 {
 	unsigned long pfn;
 	struct page *page, *head;
-	int ret = 0;
 	LIST_HEAD(source);
 	static DEFINE_RATELIMIT_STATE(migrate_rs, DEFAULT_RATELIMIT_INTERVAL,
 				      DEFAULT_RATELIMIT_BURST);
@@ -1678,7 +1677,6 @@ static void do_migrate_range(unsigned lo
 						    page_is_file_lru(page));
 
 		} else {
-			ret = -EBUSY;
 			if (__ratelimit(&migrate_rs)) {
 				pr_warn("failed to isolate pfn %lx\n", pfn);
 				dump_page(page, "isolation failed");
@@ -1692,6 +1690,7 @@ static void do_migrate_range(unsigned lo
 			.nmask = &nmask,
 			.gfp_mask = GFP_USER | __GFP_MOVABLE | __GFP_RETRY_MAYFAIL,
 		};
+		int ret;
 
 		/*
 		 * We have checked that migration range is on a single zone so
_
